automatic_generation.py

```diff
@@ -0,0 +1,105 @@
+import subprocess
+import cv2
+from simple_term_menu import TerminalMenu
+# pylint: disable-next=import-error
+from TTS.api import TTS
+import reddit_integration as ri
+import utils
+import tts
+
+BACKGROUND_VIDEO_PATH = './video-generation/public/video/backgroundVideo.mp4'
+OUTPUT_FILE_BUZZ_WORDS = 'reddit_story_crazy_reading_reddit_short_funny_sad_insane'
+RUN_OPTIONS = ['One', 'Specify number']
+
+def preview():
+    generate_videos(number_of_videos=1, run_in_preview_mode=True)
+
+def default():
+    terminal_menu = TerminalMenu(RUN_OPTIONS)
+    selected_index = terminal_menu.show()
+
+    if selected_index == 0:
+        number_of_videos = 1
+    elif selected_index == 1:
+        number_of_videos = int(input('Input number of videos to generate:\n'))
+
+    generate_videos(number_of_videos=number_of_videos)
+
+
+def generate_videos(number_of_videos, run_in_preview_mode=False):
+    log = utils.get_logger()
+    log.info('Generating %s video(s)', number_of_videos)
+
+    generated_videos = 0
+
+    # Create a tts instance early to avoid re-creation on each loop
+    tts_instance = TTS(model_name="tts_models/en/vctk/vits")
+
+    while generated_videos < number_of_videos :
+        # Get a reddit post
+        try:
+            post = ri.get_post()
+        except ImportError as error:
+            log.warning(error)
+            return
+        title = post['title']
```

Comment:
I don't have anything against it, but why are we not just sending the full post object to the different methods?